### PR TITLE
[TwigBundle] Fix Twig cache is not properly warmed

### PR DESCRIPTION
**Pull request by tucksaun**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15982 |
| License | MIT |
| Doc PR | - |

Alternative to #15034

### CODE REVIEW

File: src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheCacheWarmer.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ class TemplateCacheCacheWarmer implements CacheWarmerInterface
      * @param ContainerInterface      $container The dependency injection container
      * @param TemplateFinderInterface $finder    The template paths cache warmer
```
Comment (Tobion) [inline src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheCacheWarmer.php]: `|null` and please add explanation why null it allowed

File: src/Symfony/Bundle/TwigBundle/Tests/Functional/CacheWarmingTest.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    {
+        $this->deleteTempDir();
+    }
+
+    protected function tearDown()
+    {
+        $this->deleteTempDir();
+    }
+
+    protected function deleteTempDir()
```
Comment (Tobion) [inline src/Symfony/Bundle/TwigBundle/Tests/Functional/CacheWarmingTest.php]: should be private

**Comment by Tobion** [comment]
What I don't get is why are templates not beeing warmed if the templating component is not used? I mean we have the goal to use twig without templating, but the warmup should still compile all templates ideally, or not?

**Comment by fabpot** [comment]
:+1:

**Comment by fabpot** [comment]
@Tobion That's not the purpose of this PR but to answer your question: simply because nobody ever implemented it. In the past, it was not really needed as the templating component was mandatory. Now that it's not anymore, we need another cache warmer (I will submit a PR this week-end to fill the gap).

**Comment by Tobion** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @tucksaun.

**Pull request merged**